ffmpeg-plugin/mcm_common.c

```diff
@@ -4,156 +4,217 @@
+
+        *mc = NULL;
+    }
+
+    pthread_mutex_unlock(&mx);
+
+    return err;
+}
+
+/**
+ * Parse MCM connection parameters and fill the structure.
+ */
+int mcm_parse_conn_param(AVFormatContext* avctx, MeshConnection *conn,
+                         int kind, char *ip_addr, char *port,
+                         char *protocol_type, char *payload_type,
+                         char *socket_name, int interface_id)
+{
+    int err;
 
     /* protocol type */
     if (!strcmp(protocol_type, "memif")) {
-        param->protocol = PROTO_MEMIF;
-        snprintf(param->memif_interface.socket_path, sizeof(param->memif_interface.socket_path),
+        MeshConfig_Memif cfg;
+
+        snprintf(cfg.socket_path, sizeof(cfg.socket_path),
             "/run/mcm/mcm_memif_%s.sock", socket_name ? socket_name : "0");
-        param->memif_interface.interface_id = interface_id;
-        param->memif_interface.is_master = (type == is_tx) ? 1 : 0;
-    } else if (!strcmp(protocol_type, "udp")) {
-        param->protocol = PROTO_UDP;
-    } else if (!strcmp(protocol_type, "tcp")) {
-        param->protocol = PROTO_TCP;
-    } else if (!strcmp(protocol_type, "http")) {
-        param->protocol = PROTO_HTTP;
-    } else if (!strcmp(protocol_type, "grpc")) {
-        param->protocol = PROTO_GRPC;
-    } else {
-        param->protocol = PROTO_AUTO;
-    }
+        cfg.interface_id = interface_id;
 
-    /* payload type */
-    if (!strcmp(payload_type, "st20")) {
-        param->payload_type = PAYLOAD_TYPE_ST20_VIDEO;
-    } else if (!strcmp(payload_type, "st22")) {
-        param->payload_type = PAYLOAD_TYPE_ST22_VIDEO;
-        param->payload_codec = PAYLOAD_CODEC_JPEGXS;
-    } else if (!strcmp(payload_type, "st30")) {
-        param->payload_type = PAYLOAD_TYPE_ST30_AUDIO;
-    } else if (!strcmp(payload_type, "st40")) {
-        param->payload_type = PAYLOAD_TYPE_ST40_ANCILLARY;
-    } else if (!strcmp(payload_type, "rtsp")) {
-        param->payload_type = PAYLOAD_TYPE_RTSP_VIDEO;
+        err = mesh_apply_connection_config_memif(conn, &cfg);
+        if (err)
+            return err;
     } else {
-        av_log(avctx, AV_LOG_ERROR, "Unknown payload type\n");
-        return AVERROR(EINVAL);
+        MeshConfig_ST2110 cfg;
+
+        strlcpy(cfg.remote_ip_addr, ip_addr, sizeof(cfg.remote_ip_addr));
+        
+        if (kind == MESH_CONN_KIND_SENDER)
+            cfg.remote_port = atoi(port);
+        else
+            cfg.local_port = atoi(port);
+
+        /* transport type */
+        if (!strcmp(payload_type, "st20")) {
+            cfg.transport = MESH_CONN_TRANSPORT_ST2110_20;
+        } else if (!strcmp(payload_type, "st22")) {
+            cfg.transport = MESH_CONN_TRANSPORT_ST2110_22;
+        } else if (!strcmp(payload_type, "st30")) {
+            cfg.transport = MESH_CONN_TRANSPORT_ST2110_30;
+        } else {
+            av_log(avctx, AV_LOG_ERROR, "Unknown payload type\n");
+            return -MESH_ERR_CONN_CONFIG_INVAL;
```

Comment:
this "return" should be remained here (even if introduced other changes with return's).

Reply:
not important. I commented removal of other "returns" in this function (to replace them with 1 at the end. But here there should be no change (even then).